postal/internal/dependency_mirror.go

```diff
@@ -0,0 +1,143 @@
+package internal
+
+import (
+	"fmt"
+	"net/url"
+	"os"
+	"strings"
+)
+
+type DependencyMirrorResolver struct {
+	bindingResolver BindingResolver
+}
+
+func NewDependencyMirrorResolver(bindingResolver BindingResolver) DependencyMirrorResolver {
+	return DependencyMirrorResolver{
+		bindingResolver: bindingResolver,
+	}
+}
+
+func formatAndVerifyMirror(mirror, uri string) (string, error) {
+	mirrorURL, err := url.Parse(mirror)
+	if err != nil {
+		return "", err
+	}
+
+	uriURL, err := url.Parse(uri)
+	if err != nil {
+		return "", err
+	}
+
+	if mirrorURL.Scheme != "https" && mirrorURL.Scheme != "file" {
+		return "", fmt.Errorf("invalid mirror scheme")
+	}
+
+	mirrorURL.Path = strings.Replace(mirrorURL.Path, "{originalHost}", uriURL.Host+uriURL.Path, 1)
+	return mirrorURL.String(), nil
+}
+
+func (d DependencyMirrorResolver) FindDependencyMirror(uri, platformDir string) (string, error) {
+	mirror, err := d.findMirrorFromEnv(uri)
+	if err != nil {
+		return "", err
+	}
+
+	if mirror != "" {
+		return formatAndVerifyMirror(mirror, uri)
+	}
+
+	mirror, err = d.findMirrorFromBinding(uri, platformDir)
+	if err != nil {
+		return "", err
+	}
+
+	if mirror != "" {
+		return formatAndVerifyMirror(mirror, uri)
+	}
+
+	return "", nil
+}
+
+func (d DependencyMirrorResolver) findMirrorFromEnv(uri string) (string, error) {
+	const DefaultMirror = "BP_DEPENDENCY_MIRROR"
+	const DefaultMirrorPrefix = "BP_DEPENDENCY_MIRROR_"
```

Comment:
for clarity, wouldn't this be the non-default mirror prefix?
```suggestion
	const NonDefaultMirrorPrefix = "BP_DEPENDENCY_MIRROR_"
```